src/background.js

```diff
@@ -93,21 +94,40 @@ function init() {
 
   getMetricsUUID().then(clientUUID => {
     Metrics.init(clientUUID);
-    return getAlarms();
+    return getNextOpenAlarms();
   }).then(items => {
-    const due = Object.entries(items).filter(item => item[1].time === NEXT_OPEN);
-    const updated = {};
-    due.forEach(item => {
-      item[1].time = Date.now();
-      updated[item[0]] = item[1];
-    });
-    log('setting next open tabs to now', updated);
-    return saveAlarms(updated).then(updateWakeAndBookmarks);
+    if (items.length) {
+      browser.windows.onCreated.addListener(handleNextOpen);
+    }
   }).catch(reason => {
     log('init storage get rejected', reason);
   });
 }
 
+function handleNextOpen(window) {
+  if (!window.incognito) {
```

Comment:
Could just do `if (window.incognito) { return; }` here and drop a level of indent throughout the rest.